File: services/Whisprr.Api/Models/DTOs/Auth/AuthResponse.cs
namespace Whisprr.Api.Models.DTOs.Auth;

/// <summary>
/// Response containing authentication tokens and user info.
/// </summary>
public class AuthResponse
{
    public Guid UserId { get; set; }
    public required string Email { get; set; }
    public string? DisplayName { get; set; }
    public required string Token { get; set; }
    public required string RefreshToken { get; set; }
}

File: apps/whisprr-web-app/src/models/dtos/auth/auth-response.ts
import z from "zod";

export const AuthResponse = z.object({
  userId: z.string().uuid(),
  email: z.string().email(),
  displayName: z.string().optional(),
  token: z.string(),
  refreshToken: z.string()
});

export type AuthResponse = z.infer<typeof AuthResponse>;
